Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -43,6 +62,49 @@ class SentryProfilerSwiftTests: XCTestCase {
 #endif
     }
 
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+    func testUIEventTrackerProfiling() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        fixture.eventTracker.start()
+        SentrySDK.setCurrentHub(fixture.hub)
+
+        let exp = expectation(description: "send action finishes")
+        fixture.swizzleWrapper.swizzleSendAction({ _, _, _, _ in
+            exp.fulfill()
+        }, forKey: "first")
+
+        // based on SentryUIEventTrackerTests.test_SubclassOfUIButton_CreatesTransaction
+        fixture.swizzleWrapper.execute(action: fixture.action, target: fixture.target, sender: fixture.button, event: Fixture.TestUIEvent())
+
+        waitForExpectations(timeout: 5)
+
+        let assertionExp = expectation(description: "performed assertions")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2, execute: .init(block: {
```

Comment:
`m`: `.now() + 2` slows down the test suite and could lead to flaky tests. Why do we need it? Can we avoid it?

Reply:
we haven't yet mocked out the components for the profiler, so we need to wait a small amount of time to actually let it gather some frames to assert for later in the test case. Let's tackle improving these tests later as they are all like this.